conda_mirror/conda_mirror.py

```diff
@@ -890,6 +921,68 @@ def _validate_or_remove_package(args):
     )
 
 
+def _find_non_recent_packages(
+    packages: Dict[str, Dict[str, Any]],
+    *,
+    include: Iterable[str],
+    latest_non_dev: int,
+    latest_dev: int,
+) -> Set[str]:
+    """Computes set of package filenames that are not sufficiently recent
+
+    Parameters
+    ----------
+    packages: packages dictionary from repodata.json
+    include: package filenames to be considered
+    latest_non_dev: number of non-dev packages that are sufficiently recent to be included
+        if negative, then all non-dev packages will be included
+    latest_dev: number of dev packages that are sufficnetly recent to be included
+        if negative, then all dev packages will be included
+
+    Returns
+    -------
+    non-recent packages: Set[str]
+       Package filenames that are not sufficiently recent. This will be a subset of `include`
+    """
+
+    non_recent_packages: Set[str] = set()
+
+    if latest_non_dev >= 0 or latest_dev >= 0:
+
+        class PackageAndVersion(NamedTuple):
+            package_file: str
+            version: VersionOrder
+
+        packages_by_name: Dict[str, List[PackageAndVersion]] = {}
+        for key in include:
+            metadata = packages[key]
+            try:
+                packages_by_name.setdefault(metadata["name"], []).append(
+                    PackageAndVersion(key, VersionOrder(metadata["version"]))
+                )
+            except KeyError:
+                pass  # ignore bad entries
+
+        for curpackages in packages_by_name.values():
+            curpackages.sort(
+                key=lambda x: x.version, reverse=True
+            )  # recent versions first
+            dev_versions = [
+                p.package_file for p in curpackages if "DEV" in p.version.version[-1]
+            ]
```

Comment:
This is non-standard. There is no definition of `DEV` versions in conda. I'm happy to merge this PR without this section but I don't like to add custom interpretations into a common tool.